src/pages/en/guides/typescript.md

```diff
@@ -135,3 +137,25 @@ To see type errors in your editor, please make sure that you have the [Astro VS
 
 📚 Read more about [`.ts` file imports](/en/guides/imports/#typescript) in Astro.  
 📚 Read more about [TypeScript Configuration](https://www.typescriptlang.org/tsconfig/).
+
+## Troubleshooting
+
+### Typing multiple JSX frameworks at the same time
+
+An issue arise when using multiple JSX frameworks in the same project, as every framework require different, conflicting settings inside `tsconfig.json`. To workaround this issue, TypeScript support [different pragma comments](https://www.typescriptlang.org/docs/handbook/jsx.html#configuring-jsx) to set the JSX options on a per-file basis.
+
+The [`jsxImportSource` setting](https://www.typescriptlang.org/tsconfig#jsxImportSource) being by default `react`, all your `.jsx` files will by default be typechecked using the React definitions if installed (`@types/react`). As such, your Preact and Solid files should contain pragma comments to set the setting to the proper value:
+
+```jsx
+// For Preact
+/** @jsxImportSource preact */
+
+// For Solid
+/** @jsxImportSource solid-js */
+```
+
+Alternatively, if your project primarily use another framework, you may change the `jsxImportSource` setting to `preact` or `solid-js` and instead use a pragma comment inside your React files.
+
+### Typing Vue components when the `@types/react` package is installed
+
+Due to being declared globally, the types definition from the `@types/react` package, if installed, will mistakenly be used to typecheck `.vue` files when using [Volar](https://github.com/johnsoncodehk/volar). There's currently no reliable way to fix this, however a few solutions are discussed can be found in [this GitHub discussion](https://github.com/johnsoncodehk/volar/discussions/592)
```

Comment:
```suggestion
### Vue components are mistakenly typed by the `@types/react` package when installed

The types definition from the`@types/react` package is declared globally and therefore will be mistakenly used to typecheck `.vue` files when using [Volar](https://github.com/johnsoncodehk/volar).

**Status**:  Expected behavior.

**Solution**: There's currently no reliable way to fix this, however a few solutions and more discussion can be found in [this GitHub discussion](https://github.com/johnsoncodehk/volar/discussions/592).

```
Attempting to make this look more like the entries from our TroubleShooting page, emphasizing the status of the behaviour and solution.